xdcr/cbs_xdcr.go

```diff
@@ -147,8 +158,13 @@ func (x *CouchbaseServerXDCR) Start(ctx context.Context) error {
 	body.Add("toBucket", x.toBucket.GetName())
 	body.Add("toCluster", xdcrClusterName)
 	body.Add("replicationType", "continuous")
-	// filters all sync docs, except binary docs (attachments)
-	body.Add("filterExpression", fmt.Sprintf("NOT REGEXP_CONTAINS(META().id, \"^%s\") OR REGEXP_CONTAINS(META().id, \"^%s\")", base.SyncDocPrefix, base.Att2Prefix))
+	// if mobile setting is active, set the setting on thw replication
+	if x.MobileSetting != MobileOff {
+		body.Add("mobile", x.MobileSetting)
+	} else {
+		// filters all sync docs, except binary docs (attachments)
+		body.Add("filterExpression", fmt.Sprintf("NOT REGEXP_CONTAINS(META().id, \"^%s\") OR REGEXP_CONTAINS(META().id, \"^%s\")", base.SyncDocPrefix, base.Att2Prefix))
+	}
```

Comment:
This was a hack when I couldn't use mobile, I think xdcr without {{mobile}} is not something we need to test, or we might want a custom filter expression if we were testing unidirectional XDCR for backup/restore purposes.

Reply:
Will refactor to allow a custom filter expression to be supplied instead.